internal/resources/gitrepository/resource_git_repository.go

```diff
@@ -40,3 +55,259 @@ var gitRepositorySchema = map[string]*schema.Schema{
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			gitRepositoryReq := &gitrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdGitrepositoryGitRepositoryRequest{
+				GitRepository: &gitrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdGitrepositoryGitRepository{
+					FullName: scopedFullnameData.FullnameCluster,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterScope(d),
+				},
+			}
+
+			gitRepositoryResponse, err := config.TMCConnection.ClusterGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdGitrepositoryResourceServiceCreate(gitRepositoryReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster git repository entry, name : %s", gitRepositoryName))
+			}
+
+			UID = gitRepositoryResponse.GitRepository.Meta.UID
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			gitRepositoryReq := &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositoryGitRepositoryRequest{
+				GitRepository: &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositoryGitRepository{
+					FullName: scopedFullnameData.FullnameClusterGroup,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterGroupScope(d),
+				},
+			}
+
+			gitRepositoryResponse, err := config.TMCConnection.ClusterGroupGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdGitrepositoryResourceServiceCreate(gitRepositoryReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster group git repository entry, name : %s", gitRepositoryName))
+			}
+
+			UID = gitRepositoryResponse.GitRepository.Meta.UID
+		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
+	}
+
+	// always run
+	d.SetId(UID)
+
+	return dataSourceGitRepositoryRead(ctx, d, m)
+}
+
+func resourceGitRepositoryDelete(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	gitRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository name")
+	}
+
+	gitRepositoryNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, gitRepositoryName, gitRepositoryNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to delete Tanzu Mission Control git repository entry; Scope full name is empty")
+	}
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
```

Comment:
here, we should delete right ?

Reply:
As discussed we will not enable the feature in case of read and delete.

But I don't think we should disable it by deleting a single resource.

Reply:
Updated.